namespace-registry/registry.go

```diff
@@ -231,83 +232,78 @@ func keySignChallengeInit(ctx *gin.Context, data *registrationData) error {
 	return nil
 }
 
-type jwks struct {
-	X string `json:"x"`
-	Y string `json:"y"`
-}
-
-func jwksToEcdsaPublicKey(jwks *jwks) (*ecdsa.PublicKey, error) {
-	xBigInt, err := new(big.Int).SetString(jwks.X, 10)
-	if !err {
-		return nil, errors.New("Failed to convert jwks.x to Big Int")
-	}
-	yBigInt, err := new(big.Int).SetString(jwks.Y, 10)
-	if !err {
-		return nil, errors.New("Failed to convert jwks.y to Big Int")
+func keySignChallengeCommit(ctx *gin.Context, data *registrationData, action string) error {
+	// Parse the client's jwks as a set here
+	clientJwks, err := jwk.Parse(data.Pubkey)
+	if err != nil {
+		return errors.Wrap(err, "Couldn't parse the pubkey from the client")
 	}
 
-	clientPubkey := &ecdsa.PublicKey{
-		Curve: elliptic.P521(),
-		X:     xBigInt,
-		Y:     yBigInt,
+	if log.GetLevel() == log.DebugLevel || log.GetLevel() == log.TraceLevel {
+		// Let's check that we can convert to JSON and get the right thing...
+		jsonbuf, err := json.Marshal(clientJwks)
+		if err != nil {
+			return errors.Wrap(err, "failed to marshal the client's keyset into JSON")
+		}
+		log.Debugf("Client JWKS as seen by the registry server: %s\n", jsonbuf)
 	}
 
-	return clientPubkey, nil
-}
+	// Now iterate through the jwks and verify signatures:
```

Comment:
Why are we iterating here?  I assume it's because, if there are multiple public keys in the submitted JWKS, any one of them could have been used for the challenge, right?  If that's right, can you add a comment along those lines?

That somewhat defeats the purpose of only registering keys we can prove we have access to, right?  That means either:
1.  We should only allow the addition of one public key at a time, OR
2. We should extend the challenge to allow multiple signatures.

I think I lean toward (1).  We will need a way to add new public keys to (or remove public keys from) the namespace.  However, I think this is clearly a separate issue.  Once you expand the comment, can you please also create a new issue for adding/removing public keys from the namespace?